packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,204 @@
+
+  function handleAuth(auth) {
+    const scope = ScopeSet.fromArray(auth.credentials.scope);
+    if (! scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
+      throw error.invalidScopes('Invalid authentication scope in token');
+    }
+    const { user, email } = auth.credentials;
+    return { uid: user, email };
+  }
+
+  return [
+    {
+      method: 'GET',
+      path: '/oauth/subscriptions/plans',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken'
+        },
+        response: {
+          schema: isA.object().keys({
+            plan_id: validators.subscriptionsPlanId.required(),
+            product_id: validators.subscriptionsProductId.required(),
+            interval: isA.string().required(),
+            amount: isA.number().required(),
+            currency: isA.string().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listPlans', request);
+        handleAuth(request.auth);
+        return subhub.listPlans();
+      }
+    },
+    {
+      method: 'GET',
+      path: '/oauth/subscriptions/active',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken'
+        },
+        response: {
+          schema: isA.object().keys({
+            uid: isA.string().required(),
+            subscriptionId: validators.subscriptionsSubscriptionId.required(),
+            productName: validators.subscriptionsProductId.required(),
+            createdAt: isA.number().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.listActive', request);
+        const { uid } = handleAuth(request.auth);
+        return db.fetchAccountSubscriptions(uid);
+      }
+    },
+    {
+      method: 'POST',
+      path: '/oauth/subscriptions/active',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken'
+        },
+        validate: {
+          payload: {
+            planId: validators.subscriptionsPlanId.required(),
+            paymentToken: validators.subscriptionsPaymentToken.required()
+          }
+        },
+        response: {
+          schema: isA.object().keys({
+            subscriptionId: validators.subscriptionsSubscriptionId.required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.createSubscription', request);
+
+        const { uid, email } = handleAuth(request.auth);
+
+        await customs.check(request, email, 'createSubscription');
+
+        const { planId, paymentToken } = request.payload;
+
+        // Find the selected plan and get its product ID
+        const plans = await subhub.listPlans();
+        const selectedPlan = plans.filter(p => p.plan_id === planId)[0];
+        if (! selectedPlan) {
+          throw error.unknownSubscriptionPlan(planId);
```

Comment:
If this happens, what happens to the paymentToken, IIUC, the payment is already established and the user will be getting charged (possibly repeatedly?). Do we need to notify subhub to cancel the payment?

Reply:
As I understand it, the payment token represents a method of payment and consent to pay - i.e. not yet a completed payment.

It doesn't actually result in payment until we've successfully passed it to subhub and then subhub passes it back to the payment processor.

Reply:
> It doesn't actually result in payment until we've successfully passed it to subhub and then subhub passes it back to the payment processor.

Good, let's verify just to make sure. @bbangert can you confirm ^^^, or is there someone from IT we can verify this with?

Reply:
I mean, I can verify it insofar as this is how we've been discussing it up til now and it's now the subhub API is spec'd to work.